### PR TITLE
Add environment file

### PR DESCRIPTION
**Pull request by tswicegood**
While working on bokeh.js, I've gathered a list of packages that are needed to run examples and tests and such.  Please let me know what I'm missing.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+dependencies:
+  # Basic dependencies for examples
+  - bower
+  - dateutil
+  - gulp
+  - ipython
+  - jinja2
+  - numpy
+  - pandas
+  - python
```
Comment (tswicegood) [inline environment.yml]: Should be this be `python=2.7*`?

**Comment by mattpap** [comment]
All development dependencies are under `test` section in `meta.yaml`. Your list is way too short to do serious bokeh development. In [1] you can see how we extract this.

[1] https://github.com/bokeh/bokeh/blob/master/scripts/travis_install#L33

**Comment by tswicegood** [comment]
@mattpap cool -- didn't dig into that file to find the list.  This is basically the result in my `history` file showing all of the things I've installed trying to get JS tests working and examples.

**Comment by mattpap** [comment]
I suggest running `tests/travis/test`. This will run all examples, notebooks etc. and require most of the dependencies in `meta.yaml`. Most, because e.g. you don't need pdiff by default, just if want compare images produced by multiple invocations of our test machinery (needs special configuration options, see `.travis.yml` as an example).

**Comment by bryevdv** [comment]
@tswicegood this GH diff has gone sideways. Is there anything here we need  to pursue, or can I close this?

**Comment by bryevdv** [comment]
To add: I had @Maggie-M create a script that checks for miss dependencies that are needed for dev/test/docs but that are not part of the normal dep list. I think handling dev dependencies is really a documentation issue.

**Comment by bryevdv** [comment]
Closing this as redundant.

**Pull request closed without merging**